finesse/hardware/device.py

```diff
@@ -65,11 +65,11 @@ def get_device_types() -> dict[DeviceBaseTypeInfo, list[DeviceTypeInfo]]:
 class AbstractDevice(ABC):
     """An abstract base class for devices."""
 
-    _device_base_type_info: DeviceBaseTypeInfo
+    _device_base_type_info: ClassVar[DeviceBaseTypeInfo]
```

Comment:
What's `ClassVar` for?

Reply:
It indicates that the variable is a class variable so it should not be being changed by individual instances of `AbstractDevice`. (I just tried doing this to see what happened and `mypy` duly told me off.)

Ruff suggested this one too.